src/handlers.js

```diff
@@ -98,22 +82,23 @@ const createReportHandlers = (reportsConfig) => {
                     // value. We'll treat these as null, rather than an empty string.
                     ...fromEntries(Array.from(ctx.request.URL.searchParams.entries()).filter(([, v]) => v !== '')),
                 };
-                ctx.state.logger.push({ dbQuery, queryArgs }).log('Executing query');
-                const result = await ctx.db.query(dbQuery, queryArgs);
-                ctx.state.logger.push({ dbQuery, queryArgs, result }).log('Query result');
 
-                ctx.response.body = result;
+                const queries = Object.entries(template.dataSource.data)
+                    .map(([k, v]) => ctx.db.query(v, queryArgs)
+                        .then((result) => ({ [k]: result })));
+                const result = await Promise.all(queries);
+                ctx.response.html = template.render(Object.assign({}, ...result));
```

Comment:
I suspect probably this should go on `ctx.state.html` or similar. `ctx.state` should be reset per-request, and `ctx.response.html`, while not part of the API, could _possibly_ be part of the API in future.

Specifically, `ctx.state` is intended for this kind of data passing.

Reply:
done in PR https://github.com/mojaloop/reporting/pull/17